File: exercises/small_problems_101/easy_8/leading_substrings.rb
def substrings_at_start(string)
  results = []
  loop do 
    results.push(string)
    string = string.chop
    break if string.length == 0
  end
  results.sort
end

puts substrings_at_start('abc') == ['a', 'ab', 'abc']
puts substrings_at_start('a') == ['a']
puts substrings_at_start('xyzzy') == ['x', 'xy', 'xyz', 'xyzz', 'xyzzy']

File: frontend/javascript/exercises/list_processing/leading_substrings.js
function substringsAtStart(string) {
  return string.split('').map(function (char, index) {
    return string.substring(0, index + 1);
  });
}

console.log(substringsAtStart('abc'));   // ['a', 'ab', 'abc']
console.log(substringsAtStart('a'));     // ['a']
console.log(substringsAtStart('xyzzy')); // ['x', 'xy', 'xyz', 'xyzz', 'xyzzy']